pandas/core/arrays/datetimes.py

```diff
@@ -374,6 +374,16 @@ def _resolution(self):
     # ----------------------------------------------------------------
     # Array-like Methods
 
+    def __array__(self, dtype=None):
+        if is_object_dtype(dtype):
+            return np.array(list(self), dtype=object)
+        elif dtype == 'i8':
+            return self.asi8
+
```

Comment:
you can .astype(dtype) if not None, then coerce to ndarray

Reply:
That makes sense.  For the time being I think matching the `Index.__array__` behavior makes sense (in part because this is overriding `Index.__array__` on `DatetimeIndex`

Reply:
sgtm. I *think* we are generally ignoring dtyep here anyhow (and AFAIR dtype is None always), unless explicity passed (llke you did)